File: app/src/main/java/com/example/cykashoppinglist/mapper/ShoplistMapper.java
package com.example.cykashoppinglist.mapper;

import com.example.cykashoppinglist.entity.Item;
import com.example.cykashoppinglist.entity.ShoplistEntry;
import com.example.cykashoppinglist.util.DateParser;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class ShoplistMapper {
	public static Item toEntity(JSONObject jsonObject) throws JSONException {
		ShoplistEntry entry = new ShoplistEntry();
		entry.setDate(DateParser.parse(jsonObject.getString("date")));
		entry.setId(jsonObject.getLong("id"));
		entry.setItem(ShoplistItemMapper.toEntity(jsonObject));
		return entry;
	}

	public static List<Item> toEntity(JSONArray jsonArray) {
		List<Item> entries = new ArrayList<>();
		for (int i = 0; i < jsonArray.length(); i++) {
			try {
				JSONObject jsonObject = jsonArray.getJSONObject(i);
				entries.add(ShoplistMapper.toEntity(jsonObject));
			} catch (JSONException e) {
				e.printStackTrace();
			}
		}
		return entries;
	}
}

File: app/src/main/java/com/example/cykashoppinglist/mapper/ShoplistMapper.kt
package com.example.cykashoppinglist.mapper

import com.example.cykashoppinglist.entity.Item
import com.example.cykashoppinglist.entity.ShoplistEntry
import com.example.cykashoppinglist.util.DateParser
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

object ShoplistMapper : Mapper {
	@Throws(JSONException::class)
	override fun toEntity(jsonObject: JSONObject): Item {
		val entry = ShoplistEntry()
		entry.date = DateParser.parse(jsonObject.getString("date"))
		entry.id = jsonObject.getLong("id")
		entry.item = ShoplistItemMapper.toEntity(jsonObject)
		return entry
	}

	override fun toEntity(jsonArray: JSONArray): List<Item> {
		val entries: MutableList<Item> = ArrayList()
		for (i in 0 until jsonArray.length()) {
			try {
				val jsonObject = jsonArray.getJSONObject(i)
				entries.add(toEntity(jsonObject))
			} catch (e: JSONException) {
				e.printStackTrace()
			}
		}
		return entries
	}
}